### PR TITLE
[#718] Fixed serve errors caused by ES6 transpilation

### PR DESCRIPTION
**Pull request by Nevraeka**
Description:
closes issue #718

### CODE REVIEW

**Comment by Nevraeka** [comment]
cc @samccone

File: gulpfile.js [hunk 1]
```diff
@@ -104,6 +104,10 @@ gulp.task('styles', function() {
   return styleTask('styles', ['**/*.css']);
 });
 
+gulp.task('elements', function() {
```
Comment (robdodson) [inline gulpfile.js]: I think this task was removed in https://github.com/PolymerElements/polymer-starter-kit/commit/abf3309d87116eee85b5a00d8163ea3bbdfe37cc because we never ship external CSS files with our elements

File: gulpfile.js [hunk 2]
```diff
@@ -173,6 +177,20 @@ gulp.task('vulcanize', function() {
     .pipe($.size({title: 'vulcanize'}));
 });
 
+// Transpile all JS to ES5.
+gulp.task('js', function () {
```
Comment (robdodson) [inline gulpfile.js]: I think we only want to do this in the recipe, not the primary gulpfile. Right @samccone ?

File: gulpfile.js [hunk 3]
```diff
@@ -271,11 +319,11 @@ gulp.task('default', ['clean'], function(cb) {
   // Uncomment 'cache-config' if you are going to use service workers.
   runSequence(
     ['ensureFiles', 'copy', 'styles'],
+    ['elements', 'js'],
```
Comment (robdodson) [inline gulpfile.js]: don't think we need elements here

File: gulpfile.js [hunk 4]
```diff
@@ -214,6 +232,36 @@ gulp.task('clean', function() {
...
+    // https: true,
+    server: {
+      baseDir: ['.tmp', 'app'],
+      middleware: [historyApiFallback()]
+    }
+  });
+
+  gulp.watch(['app/**/*.html'], ['js', reload]);
+  gulp.watch(['app/styles/**/*.css'], ['styles', reload]);
+  gulp.watch(['app/elements/**/*.css'], ['elements', reload]);
```
Comment (robdodson) [inline gulpfile.js]: can remove this line

File: gulpfile.js [hunk 5]
```diff
@@ -300,9 +348,21 @@ gulp.task('deploy-gh-pages', function() {
...
+var injectDependencies = function(task, dependencies) {
+
+  if (task !== null) {
+    dependencies.forEach(function(dependency) {
+      task.dep.push(dependency);
+    });
+  }
+};
+
+var wctTestDependencies = gulp.tasks.serve.dep;
```
Comment (robdodson) [inline gulpfile.js]: Hm I thought WCT started its own local server. Are you not able to run tests with that?

File: gulpfile.js [hunk 6]
```diff
@@ -300,9 +348,21 @@ gulp.task('deploy-gh-pages', function() {
...
+
+injectDependencies(gulp.tasks[ 'wct:local' ], wctTestDependencies);
+injectDependencies(gulp.tasks[ 'wct:sauce' ], wctTestDependencies);
```
Comment (robdodson) [inline gulpfile.js]: Do you need to run serve task to test on saucelabs?

**Comment by robdodson** [comment]
Does this continue to work when you run `gulp serve:dist`?

**Comment by Nevraeka** [comment]
It works for me locally with `gulp serve: dist`

File: gulpfile.js (hunk 6 above)
Comment (Nevraeka) [inline gulpfile.js]: I'm not sure so I can remove this if we need

Comment (robdodson) [inline gulpfile.js]: Yeah it doesn't seem related to the ES2015 stuff

File: gulpfile.js (hunk 1 above)
Comment (Nevraeka) [inline gulpfile.js]: ok. For some reason I thought the elements stuff was necessary but that makes sense

**Comment by Nevraeka** [comment]
travis is breaking on testing nodejs v0.10 which I think we agreed is no longer a nodejs target for this project . Please verify @samccone . I can change the travis config to remove [this line](https://github.com/PolymerElements/polymer-starter-kit/blob/master/.travis.yml#L13) if we need the build to pass. Thoughts?

**Comment by samccone** [comment]
We do not want the changes here to be in the gulpfile since this is part of a recipe right?
The updates should be made to the es2015 recipe instead I think :)

**Comment by abdonrd** [comment]
+1 to @samccone! :)

**Comment by Nevraeka** [comment]
@samccone @abdonrd sounds good. I'll close the PR if I can and add the changes to the recipe
